SimplifiedTests/NYPLAxisLicenseServiceTests.swift

```diff
@@ -0,0 +1,204 @@
+    
+    let designatedBookInfoURL = licenseHomeDirectory
+      .appendingPathComponent(axisKeysProvider.bookFilePathKey)
+    
+    license.saveBookInfoForFetchingLicense()
+    itemDownloader.dispatchGroup.notify(queue: .global()) {
+      if
+        let data = try? Data(contentsOf: designatedBookInfoURL),
+        let jsonObject = try? JSONSerialization.jsonObject(with: data, options: .fragmentsAllowed),
+        let json = jsonObject as? [String: String],
+        let bookVaultId = json[self.axisKeysProvider.bookVaultKey],
+        bookVaultId == TestBook.book1.bookVaultId,
+        let isbn = json[self.axisKeysProvider.isbnKey],
+        isbn == TestBook.book1.isbn
+      {
+        savedInfoExpectation.fulfill()
+      }
+    }
+    
+    wait(for: [savedInfoExpectation], timeout: 10)
+  }
+  
+  func testLicenseServiceShouldDeleteDownloadedLicense() {
+    let deletionExpectation = XCTestExpectation(
+      description: "License service should be able to delete license")
+    
+    let license = createLicenseService(bookVaultId: TestBook.book2.bookVaultId,
+                                       cypher: cypherReturningBook2VaultId,
+                                       isbn: TestBook.book2.isbn)
+    
+    let licenseURL = Bundle(for: NYPLAxisLicenseServiceTests.self)
+      .url(forResource: "license-\(license.isbn)", withExtension: "json")!
+    let savedLicenseURL = licenseHomeDirectory
+      .appendingPathComponent(axisKeysProvider.desiredNameForLicenseFile)
+    if FileManager.default.fileExists(atPath: savedLicenseURL.path) {
+      try! FileManager.default.removeItem(at: savedLicenseURL)
+    }
+    
+    let licenseData = try! Data(contentsOf: licenseURL)
+    try! NYPLAssetWriter().writeAsset(licenseData, atURL: savedLicenseURL)
+    XCTAssertTrue(FileManager.default.fileExists(atPath: savedLicenseURL.path))
+    
+    license.deleteLicenseFile()
+    itemDownloader.dispatchGroup.notify(queue: .global()) {
+      if !FileManager.default.fileExists(atPath: savedLicenseURL.path) {
+        deletionExpectation.fulfill()
+      }
+    }
+    
+    wait(for: [deletionExpectation], timeout: 10)
+  }
+  
+  private func createLicenseService(bookVaultId: String,
+                                    cypher: NYPLRSACryptographing,
+                                    isbn: String) -> CustomNYPLAxisLicenseService {
+    
+    return CustomNYPLAxisLicenseService(
+      axisItemDownloader: itemDownloader, bookVaultId: bookVaultId,
+      cypher: cypher, isbn: isbn, parentDirectory: self.licenseHomeDirectory)
+  }
+  
+}
+
+private struct TestBook {
+  let isbn: String
+  let bookVaultId: String
+  
+  static let book1 = TestBook(isbn: "9781619634015", bookVaultId: "23593B59-F4A2-4722-A3D3-4DF348037757")
+  static let book2 = TestBook(isbn: "9781250109699", bookVaultId: "872D8384-7F33-4B2C-81A5-267415C0170B")
+}
+
+
+private class CustomNYPLAxisItemDownloader: NYPLAxisItemDownloader {
```

Comment:
since this is subclassing the actual downloader class, does this mean these unit tests actually reach the network? I see you have a `NYPLAxisNetworkExecutorMock` so if the downloader is using that under the hood we are good (the internet can be unpredictable and tests will start failing randomly), but I don't see how the downloader is configured with it. I'm likely missing something, just checking.

Reply:
The only time licenseService actually downloads something is when we download the license. Since we're mocking downloaded license in our tests, we're not making any real network calls.